### PR TITLE
feat(vrack-services): polling hook for orders

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-12486

| Question         | Answer
| ---------------- | ---
| Branch?          | `master` / `release/**` / `develop` 
| Bug fix?         | yes/no
| New feature?     | yes/no
| Breaking change? | yes/no
| Tickets          | Fix #... 
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/vrack-services/src/pages/index.tsx [hunk 1]
```diff
@@ -86,7 +119,32 @@ export default function ListingPage() {
         />
         {t('createVrackServicesButtonLabel')}
       </OsdsButton>
-      <Datagrid data={data.data} />
+
+      {vrackServicesDeliveringOrders?.length > 0 && (
+        <OsdsMessage className="my-5" type={ODS_MESSAGE_TYPE.success}>
+          {'Vrack Services en cours de livraison'}
```
Comment (lionel95200x) [inline packages/manager/apps/vrack-services/src/pages/index.tsx:125]: Do we need translation key for this ?

File: packages/manager/apps/vrack-services/src/api/order/hook.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  );
+  const deliveringOrderList = orderIdsWithStatus
+    .filter(({ status }) => status === 'delivering')
+    .map(({ orderId }) => orderId);
+
+  const detailedOrderList = await Promise.all(
+    deliveringOrderList.map(toDetailedOrder),
+  );
+  return (
+    detailedOrderList
```
Comment (lionel95200x) [inline packages/manager/apps/vrack-services/src/api/order/hook.ts:43]: Could you extract this to a dedicated `function` with appropriate name it's hard to understand what buisness logic you're applying here

File: packages/manager/apps/vrack-services/src/api/order/hook.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  const orderList = await getOrderList({ dateFrom: todayMinus24Hours });
+
+  const orderIdsWithStatus = await Promise.all(
+    orderList.data.map(async (orderId) => {
+      const status = await getOrderStatus(orderId);
+      return { orderId, status: status.data };
+    }),
+  );
+  const deliveringOrderList = orderIdsWithStatus
+    .filter(({ status }) => status === 'delivering')
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/api/order/hook.ts:36]: Currently it's an union type so it's already reusable

File: packages/manager/apps/vrack-services/src/api/order/hook.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        )
+      ) {
+        queryClient.invalidateQueries({
+          queryKey: queryToInvalidateOnDelivered,
+        });
+      }
+      return statusList;
+    },
+    refetchInterval: (data, query) => {
+      console.log({ data, query });
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/api/order/hook.ts:88]: https://github.com/ovh/manager/pull/10330/files#diff-c3e6cef7ec25edbc0c400c8a033f3cf062b38dfbea2a5b84f6a5e22383e0de82

done here

File: packages/manager/apps/vrack-services/src/api/order/hook.ts (hunk 2 above)
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/api/order/hook.ts:43]: https://github.com/ovh/manager/pull/10330/files#diff-c3e6cef7ec25edbc0c400c8a033f3cf062b38dfbea2a5b84f6a5e22383e0de82

done here

File: packages/manager/apps/vrack-services/src/pages/index.tsx (hunk 1 above)
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/pages/index.tsx:125]: https://github.com/ovh/manager/pull/10330/files#diff-ac58046a8aa9f6d4236000a3dd2a5d92f669379249895fdf6ca01d6ce3835fb1

fixed here

**Review by lionel95200x — APPROVED** [review]

**Pull request merged**